### DISCUSSION TITLE
Update `db-sqlite` to support Cloudflare D1

### DISCUSSION THREAD

**Discussion by sorodrigo** [payloadcms/payload#8962] [Feature Requests & Ideas]
Now that Cloudflare supports `node-js` runtime, I was able to deploy the `payload@beta` in cloudflare workers.

I started having issues with connecting to mongo because mongoose doesn't run on CF workers. I could use postgres with CF Hyperdrive but I figured that for this app I could use a smaller db so SQLite is an option.

I was investigating what would it take to connect it to CF D1 db and it seems pretty straightforward. Only change we need to do is update the types and the connect method of the `@payload/db-sqlite` package to support passing a D1 instance from the config. And then instantiate drizzle using the `D1` driver.

I already created a fork and I'm testing out this solution. Would be good to have it natively in the library.

What do y'all think? I think this could be pretty dope, PayloadCMS on the edge!

**Comment by rustygreen** [comment]
Any update on this? @sorodrigo were you able to make this work in your fork?

**Comment by BohdanK-W32** [comment]
It would be great!

**Comment by sorodrigo** [comment]
It was a nightmare trying to sync all dependencies together, because NextJS for cloudflare has its own dep needs. I decided to not complicate my life and went with AWS + Mongo. Sorry
